src/modules/blockchain/implementation/web3-service.js

```diff
@@ -33,11 +35,47 @@ class Web3Service {
         this.logger = logger;
 
         this.rpcNumber = 0;
+        this.initializeTransactionQueue(TRANSACTION_QUEUE_CONCURRENCY);
         await this.initializeWeb3();
         this.currentBlock = await this.web3.eth.getBlockNumber();
         await this.initializeContracts();
     }
 
+    initializeTransactionQueue(concurrent) {
+        this.transactionQueue = new Queue(
+            async (args, cb) => {
+                const { contractInstance, functionName, transactionArgs, future } = args;
+                try {
+                    const result = this._executeContractFunction(
+                        contractInstance,
+                        functionName,
+                        transactionArgs,
+                    );
+                    cb();
+                    future.resolve(result);
+                } catch (error) {
+                    cb();
+                    future.revert(error);
+                }
+            },
+            { concurrent },
+        );
+    }
+
+    async queueTransaction(contractInstance, functionName, transactionArgs) {
+        return new Promise((resolve, reject) => {
```

Comment:
If possible, I think it would be good to be consistent and only use async await instead of promises.